test/distributed/_tensor/test_basic_strategy.py

```diff
@@ -0,0 +1,120 @@
+# Owner(s): ["oncall: distributed"]
+
+import torch
+from torch.distributed._tensor import DeviceMesh
+from torch.distributed._tensor.ops.basic_strategy import (
+    EinsumDims,
+    gen_einsum_strategies,
+)
+
+from torch.testing._internal.common_utils import run_tests, TestCase
+from torch.testing._internal.distributed._tensor.common_dtensor import DTensorOpTestBase
+
+
+class TestEinsumDims(TestCase):
+    def test_batch_dims(self):
+        equation = "abc,abc->abc"
+        input_dims, output_dim = EinsumDims.parse_equation(equation)
+        edims = EinsumDims.parse_dims(input_dims, output_dim)
+
+        self.assertEqual(len(edims.batch_dims), 3)
+        self.assertEqual(len(edims.contracting_dims), 0)
+        self.assertEqual(len(edims.lhs_free_dims), 0)
+        self.assertEqual(len(edims.rhs_free_dims), 0)
+
+    def test_mm_dims(self):
+        equation = "mk,kn->mn"
+        input_dims, output_dim = EinsumDims.parse_equation(equation)
+        edims = EinsumDims.parse_dims(input_dims, output_dim)
+
+        self.assertEqual(len(edims.batch_dims), 0)
+        self.assertEqual(len(edims.contracting_dims), 1)
+        self.assertEqual(len(edims.lhs_free_dims), 1)
+        self.assertEqual(len(edims.rhs_free_dims), 1)
```

Comment:
would it be better to compare on the value itself instead of its length?
```
self.assertEqual(edims.batch_dims, [])
self.assertEqual(edims.contracting_dims, ['k'])
```

Reply:
sure, updated